Emby.Server.Implementations/LiveTv/Listings/SchedulesDirect.cs

```diff
@@ -779,18 +787,23 @@ public async Task<List<ChannelInfo>> GetChannels(ListingsProviderInfo info, Canc
             using var httpResponse = await Send(options, true, info, cancellationToken).ConfigureAwait(false);
             await using var stream = await httpResponse.Content.ReadAsStreamAsync(cancellationToken).ConfigureAwait(false);
             var root = await JsonSerializer.DeserializeAsync<ChannelDto>(stream, _jsonOptions, cancellationToken).ConfigureAwait(false);
+            if (root == null)
+            {
+                return new List<ChannelInfo>();
+            }
+
             _logger.LogInformation("Found {ChannelCount} channels on the lineup on ScheduleDirect", root.Map.Count);
             _logger.LogInformation("Mapping Stations to Channel");
 
-            var allStations = root.Stations ?? new List<StationDto>();
+            var allStations = root.Stations;
 
             var map = root.Map;
             var list = new List<ChannelInfo>(map.Count);
             foreach (var channel in map)
             {
                 var channelNumber = GetChannelNumber(channel);
 
-                var station = allStations.Find(item => string.Equals(item.StationId, channel.StationId, StringComparison.OrdinalIgnoreCase))
+                var station = allStations.FirstOrDefault(item => string.Equals(item.StationId, channel.StationId, StringComparison.OrdinalIgnoreCase))
```

Comment:
Why?

Reply:
`IReadOnlyList` doesn't have a `Find` implementation. I've changed it to now use the `FindIndex` from our extensions project.